src/compiler/sys.ts

```diff
@@ -15,7 +15,7 @@ namespace ts {
         createDirectory(path: string): void;
         getExecutingFilePath(): string;
         getCurrentDirectory(): string;
-        readDirectory(path: string, extension?: string, exclude?: string[]): string[];
+        readDirectory(path: string, extension?: string, exclude?: string[], depth?: number): string[];

```

Comment:
Is there only 2 depths that we need?  If so,how about searchOption: SearchOption?   
const enum SearchOption{
                TopDirectoryOnly = 0,
                AllDirectories = 1
            }